### PR TITLE
Provided additional content that might be useful.

### PR DESCRIPTION
**Pull request by e-perl-NOAA**
Provided additional content to the .qmd file to provide further detail/explanation on things that I was fuzzy on and even had to look up, additional `usethis` functions that might be useful for auto generating certain file folders and their respective files, and sections on incorporating vignettes and data.

Of course feel free to keep what brings joy/adds something and discard what doesn't.

### CODE REVIEW

File: developing_rpkgs.qmd [hunk 1]
```diff
@@ -162,14 +167,37 @@ R code is documented using [{roxygen2}](https://roxygen2.r-lib.org/). Here is an
 
 Using these examples, add documentation to the functions in your R package. Note that the `@export` tag should be used if you want users to be able to access the function directly.
```
Comment (k-doering-NOAA) [inline developing_rpkgs.qmd:177]: Cool, this is new to me! Which IDEs have this shortcut? RStudio and VScode? Or just one of them?

Comment (k-doering-NOAA) [reply developing_rpkgs.qmd:177]: If you could add this detail in the main branch, that would be great :)

Comment (e-perl-NOAA) [reply developing_rpkgs.qmd:177]: The shortcut to get the roxygen2 skeleton? It's a shortcut in the rstudio IDE. Honestly if I were to develop a package from scratch, I'd probably start in the rstudio IDE and then switch to vscode after bulk of it is created. I like the shortcuts and built-in features to actually create a package of rstudio but I like the git/github features in vscode better for pulling down a package and making edits.

Comment (k-doering-NOAA) [reply developing_rpkgs.qmd:177]: That makes a lot of sense, thanks, Elizabeth! I agree the Rstudio shortcuts are nice, but the git features and vs code are nicer

**Pull request merged**